### PR TITLE
Sort out binary installation in Makefile

### PR DESCRIPTION
**Pull request by adrianchiris**
- Align all binaries to use a predefined bin dir in project
- Add cleanup for bin dir
- modify/rename go-get-tool to go-install-tool and clean up its logic
- cleanup lint target

Signed-off-by: adrianc <adrianc@nvidia.com>

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -53,7 +54,7 @@ else
 GOBIN=$(shell go env GOBIN)
 endif
 
-GOLANGCI_LINT = $(GOBIN)/golangci-lint
+GOLANGCI_LINT = $(BIN_DIR)/golangci-lint
```
Comment (zeeke) [inline Makefile:57]: We can also get rid of [L51](https://github.com/k8snetworkplumbingwg/sriov-network-operator/pull/399/files#diff-76ed074a9305c04054cdebb9e9aad2d818052b07091de1f20cad0bbac34ffb52R51). Right?

```
# Get the currently used golang install path (in GOPATH/bin, unless GOBIN is set)
ifeq (,$(shell go env GOBIN))
GOBIN=$(shell go env GOPATH)/bin
else
GOBIN=$(shell go env GOBIN)
endif
```

Comment (adrianchiris) [reply Makefile:57]: yes this logic is no longer needed in Makefile. will remove

**Comment by SchSeba** [comment]
/lgtm

thanks!

**Comment by adrianchiris** [comment]
@zeeke @SchSeba thx for reviewing. just addressed @zeeke comment.

once CI is green and you are happy with it, please merge :)

**Review by SchSeba — APPROVED** [review]

**Pull request merged**